src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java

```diff
@@ -274,7 +300,20 @@ public final void visitToken(DetailAST blockCommentNode) {
             }
 
             if (result.getParseErrorMessage() == null) {
-                processTree(result.getTree());
+                final DetailNode javadocTree = result.getTree();
+                if (doesCheckAcceptJavadocWithNonTightHtml()) {
+                    processTree(javadocTree);
+                }
+                else {
+                    final DetailNode nonTightHtmlTag = JavadocUtils.getNonTightHtmlTag(javadocTree);
+                    if (nonTightHtmlTag == null) {
+                        processTree(javadocTree);
+                    }
+                    else if (doesCheckViolateJavadocWithNonTightHtml()) {
+                        log(nonTightHtmlTag.getLineNumber(), nonTightHtmlTag.getColumnNumber(),
+                                "javadoc.unclosedHtml", nonTightHtmlTag.getText());
```

Comment:
> This property already existed in the message.properties, I didn't add it

It should still be in a static field regardless. No one will see it inlined there.

> but it doesn't make sense to add that error to other checks since they will never violate non tight HTML and hence never throw this error.

All other javadocs will now violate this error since you are putting this log into the abstract class.
However this will now produce the problem of getting different results in different runs which is what https://github.com/checkstyle/checkstyle/issues/3340 is about.

@romani We need to rethink this error message as it will be under all javadoc checks and expand on the issue.
Should we reconsider making this a new check instead, `NonTightJavadocCheck` and only violate javadocs that have non-tight HTML? It would make it clearer if user wants this functionality or not.